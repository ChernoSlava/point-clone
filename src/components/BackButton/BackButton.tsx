import React from "react";
import { Link } from "react-router-dom";

const BackButton = () => {
  return (
    <Link to="/" style={{ textDecoration: "none", color: "blue" }}>
      Назад
    </Link>
  );
};

export { BackButton };
