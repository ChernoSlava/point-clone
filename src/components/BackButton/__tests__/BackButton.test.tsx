/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
// eslint-disable-next-line import/no-extraneous-dependencies
import { cleanup, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { BackButton } from "../BackButton";

describe("BackButton tests", () => {
  afterEach(() => {
    cleanup();
  });

  test("renders without errors", () => {
    const { container, unmount } = render(
      <MemoryRouter>
        <BackButton />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    unmount();
  });

  test("should click and go to /foo", () => {
    const { container, unmount, getByText } = render(
      <MemoryRouter>
        <BackButton to="/foo" />
        <Routes>
          <Route path="/foo" element={<div>Test foo</div>} />
          <Route path="/" element={<div>Test</div>} />
        </Routes>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    const button = getByText("Назад");

    userEvent.click(button);
    // @ts-ignore
    expect(getByText("Test foo")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    unmount();
  });
});
