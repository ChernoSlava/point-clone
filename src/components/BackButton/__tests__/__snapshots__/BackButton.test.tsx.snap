// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton tests renders without errors 1`] = `
<div>
  <a
    href="/"
    style="text-decoration: none; color: blue;"
  >
    Назад
  </a>
</div>
`;

exports[`BackButton tests should click and go to /foo 1`] = `
<div>
  <a
    href="/foo"
    style="text-decoration: none; color: blue;"
  >
    Назад
  </a>
  <div>
    Test
  </div>
</div>
`;

exports[`BackButton tests should click and go to /foo 2`] = `
<div>
  <a
    href="/foo"
    style="text-decoration: none; color: blue;"
  >
    Назад
  </a>
  <div>
    Test foo
  </div>
</div>
`;
